spam/01_spam_tutorial.py

```diff
@@ -3,17 +3,31 @@
 # # 🚀 Snorkel Intro Tutorial: Spam Detection
 
 # %% [markdown]
-# In this tutorial, we will walk through the process of using Snorkel to build a training set for classifying YouTube comments as `SPAM` or `HAM` (not spam).
+# In this tutorial, we will walk through the process of using Snorkel to build a training set for classifying YouTube comments as spam or not spam.
 # The goal of this tutorial is to illustrate the basic components and concepts of Snorkel in a simple way, but also to dive into the actual process of iteratively developing real applications in Snorkel.
 # For an overview of Snorkel, visit [snorkel.org](http://snorkel.org).
 # You can also check out the [Snorkel API documentation](https://snorkel.readthedocs.io/).
 #
 # For our task, we have access to a large amount of *unlabeled data* in the form of YouTube comments with some metadata.
-# Our goal is to train a classifier over the comment data that can predict whether a comment is spam or not.
-# To do that, we need to label our data, but doing so for real world applications can be prohibitively slow and expensive, often taking person-weeks or months of hand-labeling.
-#
-# We therefore turn to weak supervision using **_labeling functions_**, or noisy, programmatic heuristics, to assign labels to unlabeled training data efficiently.
-# We also have access to a small amount of labeled data, which we only use for evaluation purposes.
+# Our goal is to train a classifier over the comment data that can predict whether a comment is spam or not spam.
+# To do that, we need to label our data, but doing so by hand for real world applications can be prohibitively slow and expensive, often taking person-weeks or months.
+#
+# We therefore turn to weak supervision using **_labeling functions (LFs)_**: noisy, programmatic rules and heuristics that assign labels to unlabeled training data.
+# We'll dive into the Snorkel API and how we write labeling functions later in this tutorial, but as an example,
+# we can write an LF that labels data points with `"http"` in the comment text as spam since many spam
+# comments contain links:
+#
+# ```python
+# from snorkel.labeling.lf import labeling_function
+#
+# @labeling_function()
+# def lf_contains_link(x):
+#     if "http" in x.text.lower():
+#         # Return a label of SPAM if "http" in comment text
+#         return 1
```

Comment:
can we have SPAM and ABSTAIN here? given it's a new concept, 1 vs. -1 might take a while to parse along with "what will an LF look like in this case"

Reply:
went back and forth, eventually did this to underscore that it's just an int return. will it be clear enough what SPAM and ABSTAIN are?

Reply:
high level it was about seeing a real LF before even seeing the data format and fields and explanation of what the label types are...maybe a comment on "we'll explain the format in step 2..." something so people don't get stuck trying to understand LFs and data format here instead of going through the rest of the tutorial